app/knights.py

```diff
@@ -0,0 +1,22 @@
+class Knight:
+
+    def __init__(self, name: str, power: int, hp: int) -> None:
+        self.name = name
+        self.power = power
+        self.hp = hp
```

Comment:
Create `hp` and `power` as protected attributes, add for them property and set limit for `hp` (if hp < 0 ... to avoid it in `battle` function)